src/main.rs

```diff
@@ -191,11 +216,20 @@ fn main() {
     let singleline = args.is_present("singleline");
     let grandtotal = args.is_present("total");
     let verbose = args.is_present("verbose");
+    let non_recursive = args.is_present("non_recursive");
     let minsize = unbytify::unbytify(args.value_of("minsize").unwrap()).unwrap();
     let maxsize = args.value_of("maxsize").map_or(u64::max_value(),
                                                   |v| unbytify::unbytify(v).unwrap());
     let roots = args.values_of("root").unwrap();
 
+    let regex = if args.is_present("files_pat") {
+        regexp_from_filepat(args.value_of("files_pat").unwrap()) 
+    } else if args.is_present("files_pat_regex") {
+        make_regexp(args.value_of("files_pat_regex").unwrap()) 
+    } else {
+        Regex::new(r".*").unwrap()
```

Comment:
I would prefer not to match against a regex for the case without patterns. This could be done with a generic function taking a closure `Path -> bool`.

Reply:
Ok, sounds reasonable